zebra-chain/src/orchard/tree.rs

```diff
@@ -317,6 +339,46 @@ impl NoteCommitmentTree {
         }
     }
 
+    /// Returns true if the most recently appended leaf completes the subtree
+    pub fn is_complete_subtree(&self) -> bool {
+        self.inner.value().map_or(false, |x| {
+            x.position()
+                .is_complete_subtree(TRACKED_SUBTREE_HEIGHT.into())
+        })
+    }
+
+    /// Returns subtree index at [`TRACKED_SUBTREE_HEIGHT`]
+    pub fn subtree_index(&self) -> u16 {
+        (self.position() >> TRACKED_SUBTREE_HEIGHT)
+            .try_into()
+            .expect("always within u16")
+    }
+
+    /// Returns subtree root, if any
+    pub fn subtree_root(&self) -> Option<Node> {
```

Comment:
Optional: One of the benefits of using a separate type is that it can have a simple `[u8; 32]` inner representation, because we don't do any more cryptographic operations on it. This can improve performance. (Performance optimisations are optional unless there is evidence that it is very slow.)

Reply:
This will involve a bunch of renames, so it should be scheduled after the existing code has merged. I'll open a ticket for cleanups if it turns out to be a priority.